
# Project-1


## Analysing cab rides in New York to and from the Airport

In the Data Set we have to calculate the followings:-
1).Mean speed of all the rides.
2).Number of rides taken in February.
3).Number of rides where tip more than $50.

In [1]:
import pandas as pd
import numpy as np


In [2]:
df=pd.read_csv("nyc_taxis.csv")

In [3]:
df.head()

pickup_year  pickup_month  pickup_day  pickup_dayofweek  pickup_time  \
0         2016             1           1                 5            0   
1         2016             1           1                 5            0   
2         2016             1           1                 5            0   
3         2016             1           1                 5            0   
4         2016             1           1                 5            0   

   pickup_location_code  dropoff_location_code  trip_distance  trip_length  \
0                     2                      4          21.00         2037   
1                     2                      1          16.29         1520   
2                     2                      6          12.70         1462   
3                     2                      6           8.70         1210   
4                     2                      6           5.56          759   

   fare_amount  fees_amount  tolls_amount  tip_amount  total_amount  \
0         52.0          0.8          5.54       11.65         69.99   
1         45.0          1.3          0.00        8.00         54.30   
2         36.5          1.3          0.00        0.00         37.80   
3         26.0          1.3          0.00        5.46         32.76   
4         17.5          1.3          0.00        0.00         18.80   

   payment_type  
0             1  
1             1  
2             2  
3             1  
4             2

In [4]:
df.shape

(89560, 15)

1).Mean speed of all the rides.

In [5]:
#calculating speed, assuming trip_lenth ia time in seconds
speed=df.iloc[:,7]/(df.iloc[:,8]/3600)


In [6]:
speed

0        37.113402
1        38.581579
2        31.272230
3        25.884298
4        26.371542
           ...    
89555    17.194570
89556    30.101351
89557    22.299079
89558    42.415512
89559    36.904734
Length: 89560, dtype: float64

In [7]:
speed.mean()

32.24258580925573

Mean speed is 32.24258580925573


2).Number of rides takes in february.

In [8]:
rides_feb=df[df.iloc[:,1]==2]

In [9]:
rides_feb.shape[0] 

13333

Hence total rides takes in Februray is 13333.00

3).Calcuate the no.of rides with a tip greater than $50

In [10]:
df["tip_amount"].mean()

5.814489169271996

In [11]:
tip_greater=df[df["tip_amount"]>50]

In [12]:
tip_greater.shape[0]

16

So the tips greater than $50 is 16